pages/api/profiles/index.js

```diff
@@ -33,3 +34,62 @@ export async function getProfiles(options = {}) {
 
   return JSON.parse(JSON.stringify(profiles));
 }
+
+export async function getUsers(options = {}) {
+  await connectMongo();
+
+  let profiles = [];
+  const { cards } = { cards: false, ...options };
+  const fields = cards
+    ? ["username", "name", "bio", "tags", "-_id"]
+    : ["username", "name", "bio", "tags", "location", "-_id"];
+  try {
+    profiles = await Profile.find(
+      { name: { $exists: true }, isEnabled: true },
+      fields
+    );
+  } catch (e) {
+    logger.error(e, "failed loading profiles");
+    return profiles;
+  }
+
+  return JSON.parse(JSON.stringify(profiles));
```

Comment:
Is `JSON.parse(JSON.stringify())` used for deep cloning in this case? If not, I'm unsure why we need to first stringify the array and then parse it back

Reply:
There was a reason, but I can't remember why 🤔  @dan-mba do you remember why?

Reply:
All values passed from `getServerSideProps` or `getStaticProps` to a page need to be converted to JSON.

If we let Next handle this, it will fail on certain data types returned from Mongo.

By using this hack, we return an object, but in a format that Next can successfully pass along as JSON.

Reply:
Thank you Dan for explaining 👍